src/ImageClassifier/index.js

```diff
@@ -259,34 +216,15 @@ class ImageClassifier {
 }
 
 const imageClassifier = (modelName, videoOrOptionsOrCallback, optionsOrCallback, cb) => {
-  let video;
-  let options = {};
-  let callback = cb;
+  const args = handleArguments(modelName, videoOrOptionsOrCallback, optionsOrCallback, cb)
+    .require('string', 'Please specify a model to use. E.g: "MobileNet"');
 
-  let model = modelName;
-  if (typeof model !== "string") {
-    throw new Error('Please specify a model to use. E.g: "MobileNet"');
-  } else if (model.indexOf("http") === -1) {
-    model = modelName.toLowerCase();
-  }
-
-  if (videoOrOptionsOrCallback instanceof HTMLVideoElement) {
-    video = videoOrOptionsOrCallback;
-  } else if (
-    typeof videoOrOptionsOrCallback === "object" &&
-    videoOrOptionsOrCallback.elt instanceof HTMLVideoElement
-  ) {
-    video = videoOrOptionsOrCallback.elt; // Handle a p5.js video element
-  } else if (typeof videoOrOptionsOrCallback === "object") {
-    options = videoOrOptionsOrCallback;
-  } else if (typeof videoOrOptionsOrCallback === "function") {
-    callback = videoOrOptionsOrCallback;
-  }
+  const { string, video, options = {}, callback } = args;
 
-  if (typeof optionsOrCallback === "object") {
-    options = optionsOrCallback;
-  } else if (typeof optionsOrCallback === "function") {
-    callback = optionsOrCallback;
+  let model = string;
+  // TODO: I think we should delete this.
+  if (model.indexOf("http") === -1) {
+    model = model.toLowerCase();
```

Comment:
This is a good note:
* Maybe we keep this in for now, but let's do a follow up at some point reassessing the model name handling. I feel like we talked about these somewhere? 😂

Reply:
Yes I created an issue #1339 but then I closed it because I interpreted the `if` statement backwards 🤦🏻‍♀️.  I still think it should be handled a bit differently, but agreed that we should address it separately.  I think I changed the one in `ImageClassifier` in #1362